lib/improver/utilities/save.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module for saving netcdf cubes with desired attribute types."""
+
+import iris
+
+iris.FUTURE.netcdf_promote = True
+iris.FUTURE.netcdf_no_unlimited = True
+
+
+def save_netcdf(cube, filename, unlimited_dimensions=None):
+    """Save the cube provided as a NetCDF file.
```

Comment:
It seems that for most improver CLI's you're having to pass in an empty list `unlimited_dimensions=[]`
which visually seems a bit 'noisy'. At the same time you have to account for others where 'unlimited_dimensions' is set to 'None' (if I'm understanding this correctly?)

Is it possible to change the save wrapper so that you don't have to keep specifying the 'umlimited_dimensions' empty list. I thought that you could maybe use an optional '*arg' or *kwarg in the 'save_netcdf()' function. If the argument wasn't provided then the default wrapper behaviour would be to save the netCDF file with the 'unlimited_dimensions=[]'

If the argument was specified then you could check this and set 'unlimited_dimensions=None'

Reply:
Good point.  I'll test some changes around this.  If it doesn't affect the KGO I'm tempted to remove "unlimited_dimensions" completely, and just allow the iris default behaviour throughout.  This would simplify the code as I don't see any particular reason (given the current state) why we would need this functionality.